bgpd/bgp_route.c

```diff
@@ -6489,6 +6490,30 @@ void route_vty_out(struct vty *vty, struct prefix *p, struct bgp_info *binfo,
 		return;
 	}
 
+	/*
+	 * If vrf id of nexthop is different from that of prefix,
+	 * set up printable string to append
+	 */
+	if (binfo->extra && binfo->extra->bgp_orig) {
+		const char *self = "";
+
+		if (CHECK_FLAG(binfo->flags, BGP_INFO_ANNC_NH_SELF))
+			self = "=";
+
+		if (binfo->extra->bgp_orig->vrf_id == VRF_UNKNOWN)
+			snprintf(vrf_id_str, sizeof(vrf_id_str),
+				"%s%s", "@?", self);
+		else 
+			snprintf(vrf_id_str, sizeof(vrf_id_str), "@%d%s",
```

Comment:
Use `%u` to avoid printing negative VRF IDs.

The `else` statement one line above has a trailing space.